code/game/machinery/Beacon.dm

```diff
@@ -26,7 +26,8 @@
 	Beacon.syndicate = syndicate
 	Beacon.area_bypass = area_bypass
 	Beacon.cc_beacon = cc_beacon
-	hide(T.intact)
+	if(!T.transparent_floor)
+		hide(T.intact)
```

Comment:
Mmmm, this is gonna be a pain I feel. Do you know what _isn't_ affected by this issue currently? If you don't remember off the top of your head, don't worry I can look into it.

Reply:
Okay this took *way* too long to do but here:
![test 2](https://user-images.githubusercontent.com/57483089/129461329-9ebca4fd-7d4b-4de3-9e2d-337147b0e8ba.png)
It looks like a lot of the issues come specifically from spawning the objects through the game panel. Them being mapedited in or placed manually doesn't seem to cause the same problems.

Two things which *do* have problems are passive vents and unary tanks, since they both seem to be on a layer underneath the glass rather than on top of it.

Reply:
hmm, thanks for the research that went into this. It helps a _lot_, I'll get to it and fix it tomorrow.